tokio-stream/src/wrappers/stream_close.rs

```diff
@@ -0,0 +1,59 @@
+use crate::Stream;
+use pin_project_lite::pin_project;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+pin_project! {
+    /// Stream that returns None when the stream is exhausted. To be used with StreamMap to know when a stream ends.
```

Comment:
It would be useful to include an example of how to use this together with `StreamMap`. That example should probably both go on the struct, but also on the documentation for `StreamMap`.